android/src/main/java/com/reactlibrary/JudoReactNativeModule.kt

```diff
@@ -0,0 +1,669 @@
+package com.reactlibrary
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import com.facebook.react.bridge.*
+import com.judopay.*
+import com.judopay.api.error.ApiError
+import com.judopay.api.model.response.Receipt
+import com.judopay.model.*
+import com.judopay.model.Currency
+import com.judopay.model.googlepay.GooglePayAddressFormat
+import com.judopay.model.googlepay.GooglePayBillingAddressParameters
+import com.judopay.model.googlepay.GooglePayEnvironment
+import com.judopay.model.googlepay.GooglePayShippingAddressParameters
+import java.lang.Exception
+import kotlin.collections.ArrayList
+
+class JudoReactNativeModule internal constructor(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+
+    // ------------------------------------------------------------
+    // MARK: Constants
+    // ------------------------------------------------------------
+
+    companion object {
+        private const val JUDO_PAYMENT_WIDGET_REQUEST_CODE = 1
+    }
+
+    // ------------------------------------------------------------
+    // MARK: Variables
+    // ------------------------------------------------------------
+
+    private var transactionPromise: Promise? = null
+
+    private val listener: ActivityEventListener = object : BaseActivityEventListener() {
+
+        override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent) {
```

Comment:
please add a check for the `requestCode == JUDO_PAYMENT_WIDGET_REQUEST_CODE` to filter out the rest of `onActivityResult`s

Reply:
Fixed